components/brave_ads/core/internal/deprecated/confirmations/confirmation_state_manager.cc

```diff
@@ -240,6 +238,8 @@ void ConfirmationStateManager::Save() {
 
 absl::optional<OptedInInfo> ConfirmationStateManager::GetOptedIn(
     const base::Value::Dict& dict) const {
+  DCHECK(wallet_.IsValid());
```

Comment:
Should this not be after line 246 because if a user has not opted-in they should not have a wallet